src/test/java/org/disq_bio/disq/BaiMergingTest.java

```diff
@@ -0,0 +1,141 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.disq_bio.disq;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import htsjdk.samtools.BAMIndexer;
+import htsjdk.samtools.BaiEqualityChecker;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SamInputResource;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.util.RuntimeIOException;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.disq_bio.disq.impl.formats.sam.SamFormat;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(JUnitParamsRunner.class)
+public class BaiMergingTest extends BaseTest {
+
+  private Object[] parametersForTest() throws IOException {
+    String dir = "src/test/resources";
+    return java.nio.file.Files.walk(Paths.get(dir))
+        .filter(p -> p.toString().endsWith(".bam"))
+        .filter(p -> size(p) > 64 * 1024)
+        .filter(BaiMergingTest::isCoordinateSorted)
+        .toArray();
+  }
+
+  private static long size(Path p) {
+    try {
+      return Files.size(p);
+    } catch (IOException e) {
+      throw new RuntimeIOException(e);
+    }
+  }
+
+  private static boolean isCoordinateSorted(Path p) {
+    return SamReaderFactory.makeDefault()
+            .enable(SamReaderFactory.Option.INCLUDE_SOURCE_IN_RECORDS)
+            .open(p)
+            .getFileHeader()
+            .getSortOrder()
+        == SAMFileHeader.SortOrder.coordinate;
+  }
+
+  @Test
+  @Parameters
+  public void test(String inputFile) throws Exception {
+    String inputPath = getPath(inputFile);
+
+    long inputFileLength = new File(inputPath).length();
+    int splitSize = (int) (inputFileLength + 1) / 3; // three part files
+    HtsjdkReadsRddStorage htsjdkReadsRddStorage =
+        HtsjdkReadsRddStorage.makeDefault(jsc)
+            .splitSize(splitSize)
+            .validationStringency(ValidationStringency.SILENT);
+    HtsjdkReadsRdd htsjdkReadsRdd = htsjdkReadsRddStorage.read(inputPath);
+
+    String outputPath =
+        createTempPath(SamFormat.fromFormatWriteOption(ReadsFormatWriteOption.BAM).getExtension());
+    htsjdkReadsRddStorage.write(htsjdkReadsRdd, outputPath);
```

Comment:
It would be good if this asserted somehow that it actually wrote 3 parts and then merged it.  I don't see any sane way to do that though.